File: missing_num.py
def missing_num(array):
    if len(array) < 1:
        return None

    if array[0] != 1:
        return 1

    total = sum(range(array[-1] + 1)

    for num in array:
        total -= num

    if total == 0:
        return None

    return total  

File: missing_num.js
function missingNum(arr) {
    var n = arr.length+1
    sum = 0;
    expectedSum = n * (n +1)/2;

    for (var i = 0; i < n; i++) {
        sum += arr[i];
    }
    return expectedSum - sum;
}

function missingNum(arr) {
    var len = arr.length + 1;
    var sum = 0;
    var expectedSum = len * (len +1)/2;

    for (var i; i < len; i++) {
        sum += i;

    }
    return expectedSum - sum;
}